src/pages/index.tsx

```diff
@@ -3,20 +3,157 @@ import { type NextPage } from "next";
 import { Layout } from "../components/Layout";
 import { useRouter } from "next/router";
 import dynamic from "next/dynamic";
-import { Posts } from "../components/Post";
+import { PostList } from "../components/PostList";
+import { api } from "../utils/api";
+import { useState } from "react";
+import { useDebounce } from "../hooks/useDebounce";
+import { Pagination } from "../components/Pagination";
+import { PostListFiltersValues, SearchPost } from "../components/SearchPost";
+import { z } from "zod";
+import { BlogPostCategory } from "../types/blogPost";
+import qs from "qs";
+
+const POST_LIST_FILTERS_DEBOUNCE = 500;
+const POSTS_PER_PAGE = 8;
+
+const queryParamsSchema = z.object({
+  search: z.string().optional().default(""),
+  categories: z.array(z.nativeEnum(BlogPostCategory)).optional().default([]),
+  activePage: z.coerce.number().default(0),
+});
+
+const parseQueryParams = (
+  rawQueryParams: RawQueryParams
+): z.infer<typeof queryParamsSchema> => {
+  const parsedQueryParams = queryParamsSchema.safeParse({
+    categories: rawQueryParams.categories?.split(","),
+    search: rawQueryParams.search,
+    activePage: rawQueryParams.activePage,
+  });
+
+  if (parsedQueryParams.success) {
+    return parsedQueryParams.data;
+  }
+
+  return {
+    activePage: 0,
+    categories: [],
+    search: "",
+  };
+};
+
+interface RawQueryParams {
+  search?: string;
+  categories?: string;
+  activePage?: string;
+}
 
 const HomePage: NextPage = () => {
+  const rawQueryParams = qs.parse(window.location.search, {
+    ignoreQueryPrefix: true,
+  }) as RawQueryParams;
+
+  const parsedQueryParams = parseQueryParams(rawQueryParams);
+
+  const [activePage, setActivePage] = useState(parsedQueryParams.activePage);
+  const [postListFilters, setPostListFilters] = useState<PostListFiltersValues>(
+    {
+      categories: parsedQueryParams.categories,
+      search: parsedQueryParams.search,
+    }
+  );
   const router = useRouter();
+
+  const deboucedPostListFilters = useDebounce(
```

Comment:
You have a typo

`deboucedPostListFilters` => `debouncedPostListFilters`

Install `Code Spell Checker` extension for vscode, it will tell you about typos